pennylane/templates/subroutines/mod_exp.py

```diff
@@ -0,0 +1,171 @@
+        self.hyperparameters["mod"] = mod
+        super().__init__(wires=all_wires, id=id)
+
+    @property
+    def num_params(self):
+        return 0
+
+    def _flatten(self):
+        metadata = tuple((key, value) for key, value in self.hyperparameters.items())
+        return tuple(), metadata
+
+    @classmethod
+    def _unflatten(cls, data, metadata):
+        hyperparams_dict = dict(metadata)
+        return cls(**hyperparams_dict)
+
+    def map_wires(self, wire_map: dict):
+        new_dict = {
+            key: [wire_map.get(w, w) for w in self.hyperparameters[key]]
+            for key in ["x_wires", "output_wires", "work_wires"]
+        }
+
+        return ModExp(
+            new_dict["x_wires"],
+            new_dict["output_wires"],
+            self.hyperparameters["base"],
+            self.hyperparameters["mod"],
+            new_dict["work_wires"],
+        )
+
+    @property
+    def wires(self):
+        """All wires involved in the operation."""
+        return (
+            self.hyperparameters["x_wires"]
+            + self.hyperparameters["output_wires"]
+            + self.hyperparameters["work_wires"]
+        )
+
+    def decomposition(self):  # pylint: disable=arguments-differ
+
+        return self.compute_decomposition(
+            self.hyperparameters["x_wires"],
+            self.hyperparameters["output_wires"],
+            self.hyperparameters["base"],
+            self.hyperparameters["mod"],
+            self.hyperparameters["work_wires"],
+        )
+
+    @classmethod
+    def _primitive_bind_call(cls, *args, **kwargs):
+        return cls._primitive.bind(*args, **kwargs)
+
+    @staticmethod
+    def compute_decomposition(
+        x_wires, output_wires, base, mod, work_wires
+    ):  # pylint: disable=arguments-differ
+        r"""Representation of the operator as a product of other operators.
+        Args:
+            x_wires (Sequence[int]): the wires that stores the integer :math:`x`.
+            output_wires (Sequence[int]): the wires that stores the exponentiation modulo mod :math:`base^x \, \text{mod} \, mod`.
+            base (int): integer we want to exponentiate.
+            mod (int): modulo with respect to which the exponentiation is performed, default value will be ``2^len(output_wires)``
+            work_wires (Sequence[int]): the auxiliary wires to use for the exponentiation modulo :math:`mod` when :math:`mod \neq 2^{\textrm{len(output_wires)}}`
+        Returns:
+            list[.Operator]: Decomposition of the operator
+
+        **Example**
+
+        >>> qml.ModExp.compute_decomposition(x_wires = [0, 1], output_wires = [2, 3, 4], base = 3, mod = 8, work_wires = [5, 6, 7, 8, 9])
+        [ControlledSequence(Multiplier(wires=[2, 3, 4]), control=[0, 1])]
+        """
+        op_list = []
+        op_list.append(
+            qml.ControlledSequence(
+                qml.Multiplier(base, output_wires, mod, work_wires), control=x_wires
```

Comment:
I just realize that, unlike the rest of the templates, this one is placing all the intermediate QFTs through the ControlledSequence. Is it easy to create a `_mul_in_k_mod` following the idea of the other operators? If it is not straightforward let's leave it like this for now

Reply:
Looking at it carefully, several things and tests should be changed. Could you add a TODO:?